src/hooks/sdk/useMediaStreamMetrics.tsx

```diff
@@ -0,0 +1,93 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useState, useEffect } from 'react';
+import { ClientMetricReport } from 'amazon-chime-sdk-js';
+
+import { useAudioVideo } from '../../providers/AudioVideoProvider';
+
+function isValidMetric(metric: any) {
+  return typeof metric === 'number' && !Number.isNaN(metric);
+}
+
+interface MediaStreamMetrics {
+  audioPacketsSentFractionLossPercent: number | null; // Percentage of audio packets lost (1s) from client to server
+  audioPacketsReceivedFractionLossPercent: number | null; // Percentage of audio packets lost from server to client
+  availableOutgoingBandwidth: number | null;
+  availableIncomingBandwidth: number | null;
+  videoStreamMetrics: { [key: string]: { [key: string]: {[key: string]: number} } }
+}
+
+export function useMediaStreamMetrics() {
+  const audioVideo = useAudioVideo();
+  const [mediaStreamMetrics, setMetrics] = useState<MediaStreamMetrics>({
```

Comment:
Should we have this as `[name, setName]`? While reading this one might assume that `setMetrics` is setting `metrics` by convention but the actually state variable name is different. For example, you can check [here](https://github.com/aws/amazon-chime-sdk-component-library-react/blob/master/src/hooks/sdk/useBandwidthMetrics.tsx#L20) in the `useBandwidthMetrics` hook.

Reply:
Shall we change to  `[mediaStreamMetrics, setMediaStreamMetrics]`? I feel `[name, setName]` is too vague since it is acutally the state variables are all media stream metrics. Like `audioPacketsSentFractionLossPercent` and `audioPacketsReceivedFractionLossPercent` are audio metrics and `availableOutgoingBandwidth` and `availableIncomingBandwidth` are bandwidth metrics and `videoStreamMetrics`  are video stream metrics.

Reply:
Yeah sure! I just provided a template example in the comment.